### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by jeremycastillo**
Implemented jpost for register and have skeleton for login. other code was from a late rebase and git stash apply. Only other file i modified was Abstract Model and I added code that was missing from the aggregate function for search.

### CODE REVIEW

**Comment by jeremycastillo** [comment]
not sure what broke...

File: src/main/java/com/johnstarich/moviematcher/app/JsonTransformer.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.johnstarich.moviematcher.app;
 
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
```
Comment (JohnStarich) [inline src/main/java/com/johnstarich/moviematcher/app/JsonTransformer.java:4]: Unused import?

File: src/main/java/com/johnstarich/moviematcher/app/MovieMatcherApplication.java [hunk 2]
```diff
@@ -72,14 +98,17 @@ public void moviesService() {
 			System.out.println("Searched for \""+searchQuery+"\"");
 			int results = asIntOpt(request.queryParams("results")).orElse(20);
 			int page = asIntOpt(request.queryParams("page")).orElse(1);
-			return AbstractModel.search(Movie.class, searchQuery, results, page);
+			List<Movie> l = AbstractModel.search(Movie.class, searchQuery, results, page);
```
Comment (JohnStarich) [inline src/main/java/com/johnstarich/moviematcher/app/MovieMatcherApplication.java:101]: Probably should put this back the way it was

**Comment by JohnStarich** [comment]
I'm gonna go ahead and merge but only because I have patches for the issues I commented on coming in soon

**Pull request merged**